test/k8sT/Policies.go

```diff
@@ -808,3 +815,284 @@ var _ = Describe("K8sValidatedPolicyTestAcrossNamespaces", func() {
+		// Build the containers for the server pod.
+		containers = append(containers, v1.Container{
+			Name:  fmt.Sprintf("%s-container-%d", podName, port),
+			Image: "porter:1.0",
+			Env: []v1.EnvVar{
+				{
+					Name:  fmt.Sprintf("SERVE_PORT_%d", port),
+					Value: "foo",
+				},
+			},
+			Ports: []v1.ContainerPort{
+				{
+					ContainerPort: int32(port),
+					Name:          fmt.Sprintf("serve-%d", port),
+				},
+			},
+			ReadinessProbe: &v1.Probe{
+				Handler: v1.Handler{
+					HTTPGet: &v1.HTTPGetAction{
+						Path: "/",
+						Port: intstr.IntOrString{
+							IntVal: int32(port),
+						},
+						Scheme: v1.URISchemeHTTP,
+					},
+				},
+			},
+		})
+
+		// Build the Service Ports for the service.
+		servicePorts = append(servicePorts, v1.ServicePort{
+			Name:       fmt.Sprintf("%s-%d", podName, port),
+			Port:       int32(port),
+			TargetPort: intstr.FromInt(port),
+		})
+	}
+
+	By(fmt.Sprintf("Creating a server pod %s in namespace %s", podName, namespace))
+	pod, err := k.CoreV1().Pods(namespace).Create(&v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: podName,
+			Labels: map[string]string{
+				"pod-name": podName,
+			},
+		},
+		Spec: v1.PodSpec{
+			Containers:    containers,
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	svcName := fmt.Sprintf("svc-%s", podName)
+	By(fmt.Sprintf("Creating a service %s for pod %s in namespace %s", svcName, podName, namespace))
+	svc, err := k.CoreV1().Services(namespace).Create(&v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: svcName,
+		},
+		Spec: v1.ServiceSpec{
+			Ports: servicePorts,
+			Selector: map[string]string{
+				"pod-name": podName,
+			},
+		},
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	return pod, svc
+}
+
+func cleanupServerPodAndService(k *helpers.Kubectl, pod *v1.Pod, service *v1.Service) {
+	By("Cleaning up the server.")
+	err := k.CoreV1().Pods(pod.Namespace).Delete(pod.Name, nil)
+	Expect(err).To(BeNil(), "Terminating containers are not deleted after timeout")
+
+	By("Cleaning up the server's service.")
+	err = k.CoreV1().Services(service.Namespace).Delete(service.Name, nil)
+	Expect(err).To(BeNil(), "Terminating containers are not deleted after timeout")
+}
+
+func createNetworkClientPod(k *helpers.Kubectl, ns, podName string, targetService *v1.Service, dPort int) *v1.Pod {
+	pod, err := k.CoreV1().Pods(ns).Create(&v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: podName,
+			Labels: map[string]string{
+				"pod-name": podName,
+			},
+		},
+		Spec: v1.PodSpec{
+			RestartPolicy: v1.RestartPolicyNever,
+			Containers: []v1.Container{
+				{
+					Name:  fmt.Sprintf("%s-container", podName),
+					Image: "busybox",
+					Args: []string{
+						"/bin/sh",
+						"-c",
+						fmt.Sprintf("for i in $(seq 1 5); do wget -T 8 %s.%s:%d -O - && exit 0 || sleep 1; done; exit 1",
+							targetService.Name, targetService.Namespace, dPort),
+					},
+				},
+			},
+		},
+	})
+
+	Expect(err).NotTo(HaveOccurred())
```

Comment:
Expect --> ExpectWithOffset

Add a description if the expected outcome doesn't occur.

Reply:
Why `ExpectWithOffset`?